src/pkg/cli/client/byoc/baseclient.go

```diff
@@ -80,12 +80,30 @@ func NewByocBaseClient(ctx context.Context, grpcClient client.GrpcClient, tenant
 	return b
 }
 
+func GetCdImagePath(fullQualifiedImageURI string) string {
+	return pkg.Getenv("DEFANG_CD_IMAGE", fullQualifiedImageURI)
+}
+
+func ExtractImageTag(fullQualifiedImageURI string) string {
+	index := strings.LastIndex(fullQualifiedImageURI, ":")
+	if index >= 0 {
+		return fullQualifiedImageURI[index+1:]
+	}
+
+	return ""
```

Comment:
I'm tempted to just let it crash if there's no tag. It's likely a programming error, since every call to this func should get a `repo:tag` string.
```suggestion
	return fullQualifiedImageURI[index+1:]
```

Reply:
I think doing it this way may be misleading as it'll return the image string back if there is no tag in the input (i.e. index = -1 so `fullQualifiedImageURI[0:]`). The caller should handle the possible results.